### ISSUE TITLE
When the table starts in the first column of the first line the plugin adds some duplication to the table

### ISSUE THREAD

**Issue by fcrespo82** [fcrespo82/atom-markdown-table-formatter#23]
```
|Left align|Right align|Center align|
|:---------|----------:|:----------:|
|This|This|This|
|column|column|column|
|will|will|will|
|be|be|be|
|left|right|center|
|aligned|aligned|aligned|
```
is formatted as

```
| Left align | Right align | Center align |
|:-----------|------------:|:------------:|
| This       |        This |     This     |
| column     |      column |    column    |
| will       |        will |     will     |
| be         |          be |      be      |
| left       |       right |    center    |
| aligned    |     aligned |   aligned    |
|        will |     will     |
| be         |          be |      be      |
| left       |       right |    center    |
| aligned    |     aligned |   aligned    |
```

It appears that when then table is the first thing in the document that the `myIterator` runs twice.

**Comment by fcrespo82** [comment]
Changing line 75 from ```editor.backwardScanInBufferRange(@regex, range, myIterator)```
to ```editor.scanInBufferRange(@regex, range, myIterator)``` seens to fix. But needs more tests.

**Comment by lierdakil** [comment]
I will take a look. Bear in mind that I'm insanely busy right now, so it may take a day or two for me to get to this.

**Comment by fcrespo82** [comment]
Don't worry. Take your time... I'm not in a rush and I find it rather
difficult to someone use a table as the first thing in the document.

On Tue, Dec 15, 2015 at 10:36 AM Nikolay Yakimov <notifications@github.com>
wrote:

> I will take a look. Bear in mind that I'm insanely busy right now, so it
> may take a day or two for me to get to this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fcrespo82/atom-markdown-table-formatter/issues/23#issuecomment-164751291>
> .
>
-- 
Fernando Xavier de Freitas Crespo

**Comment by lierdakil** [comment]
Looks like this goes deeper than our little package. I've opened an issue on Atom's repo. Let's see what Atom devs say first.

**Comment by fcrespo82** [comment]
I thought that too. Because the ranges were being reported correctly. Let's see what they will say.

--

Fernando X. F. Crespo

**Comment by lierdakil** [comment]
So, it's a bug in Atom core after all. Fix is available, and will likely get into next minor release. https://github.com/atom/text-buffer/pull/120

That said, we don't have to scan backwards, I don't think. I've changed testing code to provide more coverage (also to save some scrolling) and added more editor-specific tests. You can find it in `editor-tests` branch. `ColumnAsFirstItem` is merged into that one already.

**Comment by fcrespo82** [comment]
@lierdakil you're great, thank you very much for your help. There's some sense of achievement finding a bug in such a great product as Atom.

I thought we should use backwards scan not to mess with marker positions after inserting the formatting spaces, but appears as Atom already takes care of this.

**Comment by fcrespo82** [comment]
Closed by #25
